File: 010/solution_010.py
import math


def is_prime(n):
    divisible = False

    if not n % 2:
        divisible = True
    else:
        for divisor in xrange(3, int(round(math.sqrt(n))) + 1, 2):
            if not n % divisor:
                divisible = True
                break

    if not divisible:
        return True
    else:
        return False


if __name__ == '__main__':
    ceiling = 2000000
    primes = [2] + [n for n in xrange(3, ceiling + 1, 2) if is_prime(n)]
    print sum(primes)

File: 010/solution_010.erl
-module(solution_010).
-export([start/0]).


is_prime(N) ->
    case N of
        1 -> false;
        2 -> true;
        3 -> true;
        _Else ->
            case N rem 2 =:= 0 of
                true -> false;
                false ->
                    Divisor = 3,
                    MaxDivisor = round(math:sqrt(N)),
                    is_prime(N, Divisor, MaxDivisor)
            end
    end.

is_prime(N, Divisor, _)
    when N rem Divisor =:= 0 ->
        false;
is_prime(N, Divisor, MaxDivisor) ->
    case Divisor >= MaxDivisor of
        true -> true;
        false -> is_prime(N, Divisor + 2, MaxDivisor)
    end.


solution(MaxPrime) ->
    lists:sum([N || N <- lists:seq(1, MaxPrime), is_prime(N)]).


start() ->
    io:format("~p~n", [solution(2000000)]).
